crates/macros/src/composition_poly.rs

```diff
@@ -75,7 +75,7 @@ impl ToTokens for CompositionPolyItem {
 					batch_query: &binius_math::RowsBatchRef<P>,
 					evals: &mut [P],
 				) -> Result<(), binius_math::Error> {
-					if batch_query.row_len() != #n_vars {
+					if batch_query.n_rows() != #n_vars {
```

Comment:
This should also do the check on `evals` length that is in `ArithCircuitPoly::batch_evaluate`:

```
		let row_len = evals.len();
		if batch_query.row_len() != row_len {
			bail!(Error::BatchEvaluateSizeMismatch {
				expected: row_len,
				actual: batch_query.row_len(),
			});
		}
```

Reply:
Yeah, I've added this check there. And also loosen the condition a bit to require at least the necessary number of variables because sometimes we pass a single row batch for evaluations to the expressions that have a subset of variables only